From ac116c73ee9cd7b463c632251c15e8e173cde8e9 Mon Sep 17 00:00:00 2001
From: Stuart Menefy <stuart.menefy@st.com>
Date: Wed, 11 Feb 2009 13:47:26 +0000
Subject: [PATCH] sh_stm: Correct PIO pin configuration for stmmac on STi7105

Input pins for the Ethernet device were incorrectly configured as
ALT_BIDIR, they should be INPUT.

Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
---
 arch/sh/kernel/cpu/sh4/setup-stx7105.c |   22 +++++++++++-----------
 1 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7105.c b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
index fbbc37b..148fdd7 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7105.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
@@ -584,11 +584,11 @@ void stx7105_configure_ethernet(int reverse_mii, int rmii_mode, int mode,
 
 	/* MIIRX_DV/MII_EXCRS */
 	stx7105_pio_sysconf(7, 4, 1, "eth");
-	pin = stpio_request_pin(7, 4, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(7, 4, "eth", STPIO_IN);
 
 	/* MIIRX_ER/MII_EXCOL */
 	stx7105_pio_sysconf(7, 5, 1, "eth");
-	pin = stpio_request_pin(7, 5, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(7, 5, "eth", STPIO_IN);
 
 	/* MIITXD[0] */
 	stx7105_pio_sysconf(7, 6, 1, "eth");
@@ -620,35 +620,35 @@ void stx7105_configure_ethernet(int reverse_mii, int rmii_mode, int mode,
 
 	/* MIIRXCLK */
 	stx7105_pio_sysconf(8, 5, 1, "eth");
-	pin = stpio_request_pin(8, 5, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(8, 5, "eth", STPIO_IN);
 
 	/* MIIRXD[0] */
 	stx7105_pio_sysconf(8, 6, 1, "eth");
-	pin = stpio_request_pin(8, 6, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(8, 6, "eth", STPIO_IN);
 
 	/* MIIRXD[1] */
 	stx7105_pio_sysconf(8, 7, 1, "eth");
-	pin = stpio_request_pin(8, 7, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(8, 7, "eth", STPIO_IN);
 
 	/* MIIRXD[2] */
 	stx7105_pio_sysconf(9, 0, 1, "eth");
-	pin = stpio_request_pin(9, 0, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(9, 0, "eth", STPIO_IN);
 
 	/* MIIRXD[3] */
 	stx7105_pio_sysconf(9, 1, 1, "eth");
-	pin = stpio_request_pin(9, 1, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(9, 1, "eth", STPIO_IN);
 
 	/* MIITXCLK */
 	stx7105_pio_sysconf(9, 2, 1, "eth");
-	pin = stpio_request_pin(9, 2, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(9, 2, "eth", STPIO_IN);
 
 	/* MIICOL */
 	stx7105_pio_sysconf(9, 3, 1, "eth");
-	pin = stpio_request_pin(9, 3, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(9, 3, "eth", STPIO_IN);
 
 	/* MIICRS */
 	stx7105_pio_sysconf(9, 4, 1, "eth");
-	pin = stpio_request_pin(9, 4, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(9, 4, "eth", STPIO_IN);
 
 	/* MIIPHYCLK */
 	/* Not implemented in cut 1 (DDTS GNBvd69906) - clock never output */
@@ -659,7 +659,7 @@ void stx7105_configure_ethernet(int reverse_mii, int rmii_mode, int mode,
 
 	/* MIIMDINT */
 	stx7105_pio_sysconf(9, 6, 1, "eth");
-	pin = stpio_request_pin(9, 6, "eth", STPIO_ALT_BIDIR);
+	pin = stpio_request_pin(9, 6, "eth", STPIO_IN);
 
 	platform_device_register(&stx7105eth_device);
 }
-- 
1.5.6.3

